server/fleet/activities.go

```diff
@@ -802,3 +803,82 @@ func (a ActivityTypeEditedMacosProfile) Documentation() (activity, details, deta
   "team_name": "Workstations"
 }`
 }
+
+// LogRoleChangeActivities logs activities for each role change, globally and one for each change in teams.
+func LogRoleChangeActivities(ctx context.Context, ds Datastore, adminUser *User, oldGlobalRole *string, oldTeamRoles []UserTeam, user *User) error {
+	if user.GlobalRole != nil && (oldGlobalRole == nil || *oldGlobalRole != *user.GlobalRole) {
+		if err := ds.NewActivity(
+			ctx,
+			adminUser,
+			ActivityTypeChangedUserGlobalRole{
+				UserID:    user.ID,
+				UserName:  user.Name,
+				UserEmail: user.Email,
+				Role:      *user.GlobalRole,
```

Comment:
Should we not include OldRole here as well for the case of `*oldGlobalRole != *user.GlobalRole` ?

Reply:
Sorry. I don't follow.

`*user.GlobalRole` is the new role that should be logged as an activity.

Reply:
in line 831 you included `OldRole`
Should it go here as well?

Reply:
Ah, right, it depends on the activity, the one that needs the old role is an activity for the deletion of a role.

Reply:
Good catch! Added in 9bff202bc.